### PR TITLE
New Presence Score Saber

### PR DESCRIPTION
**Pull request by joerkig**
New Presence for [Scoresaber.com](https://Scoresaber.com)
0a4r95
pu0zv6
mrjs09
2p5cf9
nhy6bl
7bdxfn
lnoumw

### CODE REVIEW

**Comment by joerkig** [comment]
Sorry about closing and reopening I kinda panicked for no reason ;-;

File: websites/S/Score Saber/presence.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    else
+      presenceData.details = "Browsing Leaderboards";
+
+  if (!time) {
+    delete presenceData.startTimestamp;
+  }
+
+  if (!buttons) {
+    delete presenceData.buttons;
+  }
```
Comment (ririxidev) [inline websites/S/Score Saber/presence.ts:85]: you don't need brackets here
```suggestion
  if (!time) delete presenceData.startTimestamp;

  if (!buttons) delete presenceData.buttons;
```

**Review by ririxidev — CHANGES_REQUESTED** [review]
could you add what difficulty user is watching currently?

File: websites/S/Score Saber/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    "logo": "https://i.imgur.com/paPCAuE.png",
+    "thumbnail": "https://i.imgur.com/Lop0qlV.jpg",
+    "color": "#ffde1a",
+    "tags": [
+      "scoresaber", 
+      "beatsaber", 
+      "gaming"
+    ],
+    "category": "games",
+    "readLogs": false,
```
Comment (Bas950) [inline websites/S/Score Saber/dist/metadata.json:25]:
```suggestion
```
It's false by default

File: websites/S/Score Saber/presence.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    startTimestamp: browsingStamp
+  };
+
+  if (document.location.pathname.includes("/global"))
+    if (document.location.href.includes("?search="))
+      presenceData.details = "Searching Users",
+      presenceData.state = document
+      .querySelector("input.input")
+      .getAttribute("value");
+    else if (document.location.href.includes("?country="))
```
Comment (Bas950) [inline websites/S/Score Saber/presence.ts:22]:
```suggestion
    if (document.location.href.includes("?search=")) {
      presenceData.details = "Searching Users";
      presenceData.state = document
      .querySelector("input.input")
      .getAttribute("value");
    } else if (document.location.href.includes("?country="))
```

File: websites/S/Score Saber/presence.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      presenceData.smallImageText = document
+      .querySelector("li.is-active > a > span")
+      .textContent,
+      presenceData.buttons = [
+          {
+            label: "View Page",
+            url: document.location.href
+          }
+        ];
+    else
```
Comment (Bas950) [inline websites/S/Score Saber/presence.ts:50]:
```suggestion
    if (document.location.pathname.includes("/leaderboard/")) {
      presenceData.details = "Viewing Leaderboard";
      presenceData.state = document
      .querySelector("h4 > a")
      .textContent;  
      presenceData.smallImageKey = document
      .querySelector("li.is-active > a > span")
      .textContent.toLowerCase().replace("+","_");
      presenceData.smallImageText = document
      .querySelector("li.is-active > a > span")
      .textContent;
      presenceData.buttons = [
          {
            label: "View Page",
            url: document.location.href
          }
        ];
    } else
```

File: websites/S/Score Saber/presence.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    presenceData.smallImageText = document
+    .querySelector("li.is-active > a > span")
+    .textContent,
+    presenceData.buttons = [
+      {
+        label: "View Page",
+        url: document.location.href
+      }
+    ];
+  else if (document.location.pathname.includes("/u/"))
```
Comment (Bas950) [inline websites/S/Score Saber/presence.ts:71]:
```suggestion
  else if (document.location.pathname.includes("/ranking/request")) {
    presenceData.details = "Viewing Rank Request";
    presenceData.state = document
    .querySelector("h4.title.is-5.songInfos > a")
    .textContent;
    presenceData.smallImageKey = document
    .querySelector("li.is-active > a > span")
    .textContent.toLowerCase().replace("+","_");
    presenceData.smallImageText = document
    .querySelector("li.is-active > a > span")
    .textContent;
    presenceData.buttons = [
      {
        label: "View Page",
        url: document.location.href
      }
    ];
  } else if (document.location.pathname.includes("/u/")) {
```

File: websites/S/Score Saber/presence.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      }
+    ];
+  else if (document.location.pathname == ("/"))
+    if (document.location.href.includes("?search="))
+      presenceData.details = "Searching Leaderboards",
+      presenceData.state = document
+      .querySelector("input.input")
+      .getAttribute("value");
+    else
+      presenceData.details = "Browsing Leaderboards";
```
Comment (Bas950) [inline websites/S/Score Saber/presence.ts:89]:
```suggestion
    presenceData.details = "Viewing User";
    presenceData.state = document
    .querySelector("h5.title.is-5 > a")
    .textContent;
    presenceData.buttons = [
      {
        label: "View Page",
        url: document.location.href
      }
    ];
  } else if (document.location.pathname == ("/")) {
    if (document.location.href.includes("?search=")) {
      presenceData.details = "Searching Leaderboards";
      presenceData.state = document
      .querySelector("input.input")
      .getAttribute("value");
    } else presenceData.details = "Browsing Leaderboards";
  }
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by ririxidev — APPROVED** [review]
i guess lgtm

**Review by Bas950 — APPROVED** [review]

**Pull request merged**